### PR TITLE
Add features to limit comments and delete all comments from storage.

### PR DESCRIPTION
**Pull request by mes712**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -38,16 +36,19 @@
...
 
    @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query("Comment").addSort(TIME_PROP, SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
+    int max = Integer.parseInt(request.getParameter(MAX_COMMENT_PROP));
     ImmutableList<Comment> comments = 
         Streams.stream(results.asIterable())
             .map(entity -> (makeComment(entity)))
```
Comment (christophervo-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: you can use the .map(DataServlet::makeComment) syntax instead, which is preferable style-wise but logically equivalent to what you have here. This syntax can be used in general for .map() for single-argument methods. "DataServlet" refers to the class the method is defined in, while "makeComment" is the name of the method.

A bit more on this syntax as an FYI: it only works on single-argument methods where the argument type is equivalent to the type of the "object" reference in our stream (eg: "entity" has type Entity and makeComment will only take a single Entity, so we can use this syntax. If our makeComment took multiple arguments (say an Entity and a String), then we can't use this syntax)

File: portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.google.sps.servlets;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
```
Comment (christophervo-g) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java:3]: nit: blank line to separate static and non-static imports; same for DataServlet.java (I can't comment on that unedited line for some reason)

File: portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+@WebServlet("/delete-data")
+public class DeleteServlet extends HttpServlet {
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Comment");
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query); 
+        ImmutableList<Key> keys = 
+            Streams.stream(results.asIterable())
+                .map(entity -> entity.getKey())
```
Comment (christophervo-g) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java]: nit: similar as above comment, you can do .map(Entity::getKey)

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -32,10 +32,17 @@ function changeCatPhoto() {
 }
 
 /* Displays all comments left on page. */
-async function getComments() {
-  const response = await fetch('/data').then(response => response.json())
+async function getComments(max) {
+  document.getElementById('comments-section').innerHTML = "";
```
Comment (christophervo-g) [inline portfolio/src/main/webapp/script.js]: nit: string literals should use single quotes

https://google.github.io/styleguide/jsguide.html#features-strings-use-single-quotes

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -32,10 +32,17 @@ function changeCatPhoto() {
...
+  const response = await fetch('/data?max-comments=' + max).then(response => response.json())
   for (const comment of response) {
       document.getElementById('comments-section').innerHTML +=
-        comment['commentText'] + " --" + comment['displayName'] + "<br>";
+        comment["commentText"] + " --" + comment["displayName"] + "<br>";
```
Comment (christophervo-g) [inline portfolio/src/main/webapp/script.js]: Is there a reason this changed to using double quotes? In JS, string literals should use single quotes.

https://google.github.io/styleguide/jsguide.html#features-strings-use-single-quotes

**Review by christophervo-g — APPROVED** [review]
LGTM

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -32,10 +32,17 @@ function changeCatPhoto() {
...
 /* Displays all comments left on page. */
-async function getComments() {
-  const response = await fetch('/data').then(response => response.json())
+async function getComments(max) {
+  document.getElementById('comments-section').innerHTML = '';
+  const response = await fetch('/data?max-comments=' + max).then(response => response.json())
   for (const comment of response) {
       document.getElementById('comments-section').innerHTML +=
-        comment['commentText'] + " --" + comment['displayName'] + "<br>";
+        comment['commentText'] + " --" + comment['displayName'] + '<br>';
```
Comment (linhnguye-g) [inline portfolio/src/main/webapp/script.js]: nit: the " ---" should be single quotes as well

**Review by linhnguye-g — APPROVED** [review]
LGTM

**Pull request merged**